src/hash/blake/mod.rs

```diff
@@ -290,15 +288,25 @@ where
     let digest = if Felt::IS_CANONICAL {
         blake3::hash(E::elements_as_bytes(elements))
     } else {
-        let base_elements = E::as_base_elements(elements);
-        let blen = base_elements.len() << 3;
+        let mut hasher = blake3::Hasher::new();
```

Comment:
could you add to the documentation the expected usage pattern? What you mentioned before:

> > Our use case is somewhat different - we need to hash a large number of inputs where each input is smaller than 1KB (i.e., about 100 field elements).